packages/e2e/test/blockfrost/queryTransactions.test.ts

```diff
@@ -0,0 +1,304 @@
+          Cardano.PaymentAddress('2cWKMJemoBai9J7kVvRTukMmdfxtjL9z7c396rTfrrzfAZ6EeQoKLC2y1k34hswwm4SVr'),
+          Cardano.PaymentAddress('addr_test1vr8nl4u0u6fmtfnawx2rxfz95dy7m46t6dhzdftp2uha87syeufdg'),
+          Cardano.PaymentAddress(
+            'addr_test1qph5x6uahxhxyvtqatzj77sjtjmdjycemt5ncjuj2r4e' +
+              'yflkdap42xncd6cazjarce6jh8mx52fcf8ugststvyklj70qhzhe9h'
+          )
+        ],
+        pagination: { limit: 20, startAt: 0 }
+      });
+      expect(response.totalResultCount).toBeGreaterThanOrEqual(52);
+
+      expect(
+        response.pageResults.find(
+          (tx) => tx.id === Cardano.TransactionId('2822d491a890b40cd2a22003b81a97f63c2b8c373b1b0b8dfa1598739fe34c06')
+        )
+      ).toBeDefined();
+
+      expect(
+        response.pageResults.find(
+          (tx) => tx.id === Cardano.TransactionId('667f714ee9d9975ca4fa0f5451e006d3dafcdafb7342fe288ebcaf17c100a996')
+        )
+      ).toBeDefined();
+    });
+    it('Shelley address not used - no transactions', async () => {
+      const response = await chainHistoryProvider.transactionsByAddresses({
+        addresses: [Cardano.PaymentAddress('addr_test1vrfxjeunkc9xu8rpnhgkluptaq0rm8kyxh8m3q9vtcetjwshvpnsm')],
+        pagination: { limit: 20, startAt: 0 }
+      });
+      expect(response.totalResultCount).toBe(0);
+    });
+    it.skip('queries successfully invalid transaction (script failure)', async () => {
+      // @todo find a script failure in preprod
+      const response = await chainHistoryProvider.transactionsByAddresses({
+        addresses: [Cardano.PaymentAddress('addr_test1vr8nl4u0u6fmtfnawx2rxfz95dy7m46t6dhzdftp2uha87syeufdg')],
+        pagination: { limit: 20, startAt: 0 }
+      });
+
+      const invalidTx = response.pageResults.find(
+        (tx) => tx.id === Cardano.TransactionId('FHnt4NL7yPYH2vP2FLEfH2pt3K6meM7fgtjRiLBidaqpP5ogPzxLNsZy68e1KdW')
+      );
+      expect(invalidTx).toBeDefined();
+      expect(invalidTx?.body.outputs.length).toEqual(0);
+    });
+
+    it('stake address throws error', async () => {
+      expect(() =>
+        Cardano.PaymentAddress('stake_test1ur676rnu57m272uvflhm8ahgu8xk980vxg382zye2wpxnjs2dnddx')
+      ).toThrowError();
+    });
+
+    it('returns transactions starting from blockRange param', async () => {
+      const address = Cardano.PaymentAddress(
+        'addr_test1qpvqf0y9sgpn92crff8cxrl95s0veay4gude7pgdn0tlvwv5j9agwuw3dfhsan7u2wyzm0tfwyqstcrsjwrl4y6chjvsneezuv'
+      );
+      const response = await chainHistoryProvider.transactionsByAddresses({
+        addresses: [address],
+        blockRange: { lowerBound: Cardano.BlockNo(3_348_548) },
+        pagination: { limit: 20, startAt: 0 }
+      });
+
+      expect(
+        response.pageResults.find(
+          (tx) => tx.id === Cardano.TransactionId('78ed92583cd879c56e4dd49c06af303aa52e127074eac1123f727ed7eef36084')
+        )
+      ).toBeUndefined();
+
+      expect(response.totalResultCount).toBeGreaterThanOrEqual(3);
```

Comment:
Assertion like this would be great here
```suggestion
      expect(response.totalResultCount).toBeGreaterThanOrEqual(3);
      expect(response.every(blockHeightIsGTE(3_348_548))).toBe(true);
```

Reply:
Done in 1e29a184d5fcd8326bf85d9c0635d7c122633541